wordCount/wordCount.go

```diff
@@ -0,0 +1,80 @@
+package wordcount
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	workerpool "../workerPool"
+)
+
+//FileChan struct
+type FileChan struct {
+	FileName string
+	Channel  chan map[string]int
+}
+
+//RunJob implement for worker job
+func (f FileChan) RunJob() {
+	countedMap := wordCount(f.FileName)
+	f.Channel <- countedMap
+}
+
+// ListAllTxt return all *.txt in folder and sub-folder
+func ListAllTxt(pathToFolder string) []string {
+	var files []string
+
+	err := filepath.Walk(pathToFolder, func(path string, info os.FileInfo, err error) error {
+		if strings.Contains(path, ".txt") {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	return files
+}
+
+// readFile return the context of file
+func readFile(pathToFile string) string {
+	data, err := ioutil.ReadFile(pathToFile)
+	if err != nil {
+		fmt.Println(pathToFile, "File reading error", err)
+		return ""
+	}
+	return string(data)
+}
+
+// wordCount count exit words in the file
+func wordCount(pathToFile string) map[string]int {
```

Comment:
Try not to put file path as a func param, let abstract it by io.Reader / io.Writer. It will be easier to make unit tests.